R/bambu.R

```diff
@@ -136,24 +136,48 @@
 #'     genome = fa.file,  discovery = TRUE, quant = TRUE)
 #' @export
 bambu <- function(reads, annotations = NULL, genome = NULL, NDR = NULL,
-    opt.discovery = NULL, opt.em = NULL, rcOutDir = NULL, discovery = TRUE, 
-    quant = TRUE, stranded = FALSE,  ncore = 1, yieldSize = NULL,  
+    mode = NULL, opt.discovery = NULL, opt.em = NULL, rcOutDir = NULL, discovery = TRUE, 
+    assignDist = TRUE, quant = TRUE, stranded = FALSE,  ncore = 1, yieldSize = NULL,  
     trackReads = FALSE, returnDistTable = FALSE, lowMemory = FALSE, 
-    fusionMode = FALSE, verbose = FALSE) {
+    fusionMode = FALSE, verbose = FALSE, demultiplexed = FALSE, spatial = NULL, quantData = NULL,
+    sampleNames = NULL, cleanReads = FALSE, dedupUMI = FALSE, barcodesToFilter = NULL, clusters = NULL) {
+    message(paste0("Running Bambu-v", "3.3.0"))
+    if(!is.null(mode)){
+        if(mode == "bulk"){
+            lowMemory = TRUE
+        }
+        if(mode == "multiplexed"){
+            if(is.null(demultiplex)) demultiplex = TRUE
```

Comment:
```suggestion
            if(is.null(demultiplex)) demultiplexed = TRUE
```

Reply:
is it better to default it to FALSE when mode is "multiplexed"?
then give option for providing demultiplexed bam file